desc/objectives/_free_boundary.py

```diff
@@ -754,6 +754,283 @@ def _print(fmt, fmax, fmin, fmean, norm, units):
+        field_grid=None,
+        vacuum=False,
+        name="Quadratic flux",
+    ):
+        self._src_grid = src_grid
+        self._eval_grid = eval_grid
+        self._eq = eq
+        self._s = s
+        self._q = q
+        self._ext_field = ext_field
+        self._field_grid = field_grid
+        self._vacuum = vacuum
+        things = [ext_field]
+        super().__init__(
+            things=things,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self._eq
+
+        if self._src_grid is None:
+            src_grid = LinearGrid(
+                rho=np.array([1.0]),
+                M=15,
+                N=15,
+                NFP=int(eq.NFP),
+                sym=False,
+            )
+            self._src_grid = src_grid
+        else:
+            src_grid = self._src_grid
+
+        if self._eval_grid is None:
+            eval_grid = LinearGrid(
+                rho=np.array([1.0]),
+                M=eq.M_grid,
+                N=eq.N_grid,
+                NFP=int(eq.NFP),
+                sym=False,
+            )
+            self._eval_grid = eval_grid
+        else:
+            eval_grid = self._eval_grid
+        if self._s is None:
+            k = min(src_grid.num_theta, src_grid.num_zeta)
+            self._s = k // 2 + int(np.sqrt(k))
+        if self._q is None:
+            k = min(src_grid.num_theta, src_grid.num_zeta)
+            self._q = k // 2 + int(np.sqrt(k))
+
+        try:
+            interpolator = FFTInterpolator(eval_grid, src_grid, self._s, self._q)
+        except AssertionError as e:
+            warnings.warn(
+                "Could not built fft interpolator, switching to dft method which is"
+                " much slower. Reason: " + str(e)
+            )
+            interpolator = DFTInterpolator(eval_grid, src_grid, self._s, self._q)
+
+        self._constants = {"interpolator": interpolator}
+        self._data_keys = [
+            "R",
+            "zeta",
+            "Z",
+            "n_rho",
+            "K_vc",
+            "phi",
+        ]
+        self._args = get_params(
+            self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=False,
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        src_profiles = get_profiles(self._data_keys, obj=eq, grid=src_grid)
+        src_transforms = get_transforms(self._data_keys, obj=eq, grid=src_grid)
```

Comment:
all the source stuff (grid, profiles, transforms, interpolator etc) are only needed if `vacuum=False` so we should move it under the conditional so we don't calculate a bunch of stuff that's not needed.